package Test;

// 1
message Family
{
    required Identity identity = 1;
    repeated Person person = 2;
    optional string surname = 3;
}

// 1.1
message Identity
{
    required int32 id = 1;
    optional string postcode = 2;
    repeated string house = 3;
    optional bool is_teenager = 4;
    optional double test1 = 5;
    optional int64 test2 = 6;
}

// 1.2
message Person
{
    required int32 id = 1;
    required string name = 2;
    repeated string email = 3;
    repeated Trade trade = 4;
    optional Valuable valuable = 5;
}

// 1.2.4
message Trade
{
    required string trade_name = 1;
    required int32 trade_value = 2;
}

// 1.2.5
message Valuable
{
    repeated Goods goods = 1;
}

// 1.2.5.1
message Goods
{
    required string goods_name = 1;
    required int32 goods_value = 2;
}


